controllers/aggregated_clusteroperator_controller.go

```diff
@@ -0,0 +1,144 @@
+)
+
+type AggregatedClusterOperatorReconciler struct {
+	client.Client
+	DiscoveryClient discovery.DiscoveryInterface
+	Configv1Client  configv1client.ConfigV1Interface
+}
+
+const aggregateCOName = "platform-operators-aggregated"
+
+// TODO kubebuilder markers
+
+func (a *AggregatedClusterOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := logr.FromContext(ctx)
+	log.Info("reconciling request", "req", req.NamespacedName)
+	defer log.Info("finished reconciling request", "req", req.NamespacedName)
+
+	if req.Name != aggregateCOName {
+		return ctrl.Result{}, nil
+	}
+
+	// Create a builder to update the CO status
+	builder := aggregatedco.NewBuilder()
+	// Create a writer to write to the CO status
+	writer := aggregatedco.NewWriter(a.DiscoveryClient, a.Configv1Client)
+
+	aggregatedCO := &openshiftconfigv1.ClusterOperator{}
+	if err := a.Get(ctx, req.NamespacedName, aggregatedCO); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	defer func() {
+		if err := writer.UpdateStatus(aggregatedCO, builder.GetStatus()); err != nil {
+			log.Error(err, "error updating CO status")
+		}
+	}()
+
+	// Set the default CO status conditions: Progressing True, Degraded False, Available False
+	builder.WithProgressing(openshiftconfigv1.ConditionTrue, "")
+	builder.WithDegraded(openshiftconfigv1.ConditionFalse)
+	builder.WithAvailable(openshiftconfigv1.ConditionFalse, "", "")
+
+	POList := &platformv1alpha1.PlatformOperatorList{}
+	if err := a.List(ctx, POList); err != nil {
+		log.Error(err, "error listing platformoperators")
+		return ctrl.Result{}, err
+	}
+
+	if len(POList.Items) == 0 {
+		// No POs on cluster, everything is fine
+		builder.WithAvailable(openshiftconfigv1.ConditionTrue, "", "No POs Found")
+		return ctrl.Result{}, nil
+	}
+
+	statusErrorCheck := a.inspectPlatformOperators(POList)
+	if statusErrorCheck != nil {
+		// One of the POs is in an error state
+		// Update the Aggregated CO with the information on the failed PO
+		builder.WithDegraded(openshiftconfigv1.ConditionTrue)
+		builder.WithAvailable(openshiftconfigv1.ConditionFalse, utilerror.NewAggregate(statusErrorCheck.FailingErrors).Error(), "PO In An Error State")
+		return ctrl.Result{}, nil
+	}
+
+	builder.WithAvailable(openshiftconfigv1.ConditionTrue, "All POs in a successful state", "POs Are Healthy")
+
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (a *AggregatedClusterOperatorReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// TODO: Setup predicate filter to only watch the Aggregate CO and ignore other COs
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&openshiftconfigv1.ClusterOperator{}).
+		Watches(&source.Kind{Type: &platformv1alpha1.PlatformOperator{}}, handler.EnqueueRequestsFromMapFunc(util.RequeueBundleDeployment(mgr.GetClient()))).
+		Complete(a)
+}
+
+type POStatusErrors struct {
+	FailingPOs    []*platformv1alpha1.PlatformOperator
```

Comment:
It does not seem that we are using this property on `POStatusErrors` in the code anywhere. Is there value in storing it to memory that I'm not seeing? Is this use preparing for a use case in the future?

Reply:
Potentially, yes. I can slim down the data structure but I imagine in the future we would have more robust status messaging